pallets/living-assets-ownership/src/functions.rs

```diff
@@ -1,22 +1,54 @@
 //! Contains helper and utility functions of the pallet
 use super::*;
-use frame_support::{ensure, sp_runtime::DispatchResult};
+use frame_support::sp_runtime::{
+	traits::{CheckedAdd, One},
+	DispatchResult,
+};
 
 impl<T: Config> Pallet<T> {
-	/// See [Self::create_collection]
-	pub fn do_create_collection(
-		collection_id: T::CollectionId,
-		who: T::AccountId,
-	) -> DispatchResult {
-		ensure!(
-			!OwnerOfCollection::<T>::contains_key(collection_id),
-			Error::<T>::CollectionAlreadyExists
-		);
+	/// Attempts to create a new collection
+	///
+	/// This function is intended to be used as the implementation for the
+	/// [Self::create_collection] public interface. It first retrieves the current
+	/// collection count to use as the new collection's ID. It then inserts a new
+	/// entry into the `OwnerOfCollection` map, mapping the new collection's ID to
+	/// the owner's account ID.
+	///
+	/// After this, the function attempts to increment the collection counter by 1.
+	/// If this operation would result in an overflow, the function returns early
+	/// with an [Error::CollectionIdOverflow].
+	///
+	/// Finally, if all operations were successful, the function emits a
```

Comment:
I think it was agreed to not assign ownership to a collection, since it's not clear what privileges may the owner have (unlike in the evochain, where the owner can mint & evolve).

Also, should it return the id of the created collection? Otherwise, if not returned, is the caller of this method forced to listen to the event?

Reply:
Yes, but this collection is about create collection to return the id of the created collection